File: CPP/SZNOI 语法百题/d014 判断直角三角形/d014 判断直角三角形.cpp
#include <iostream>
using namespace std;
int main()
{
	int a, b, c;
	cin >> a >> b >> c;
	if (a*a + b*b == c*c || a*a + c*c == b*b || b*b + c*c == a*a)
	{
		cout << "Yes!";
	}
	else
		if (a + b <= c || a + c <= b || b + c <= a)
			cout << "Error!";
		else cout << "No!";
	return 0;
}

File: PASCAL/pascal/d014 判断直角三角形/d014 判断直角三角形.pas
var
a,b,c,d,e:integer;
yes,wr:boolean;
begin
readln(a,b,c);
if a+b<=c then writeln('Error!')
else begin
if a>b then
 begin
  d:=a;
  a:=b;
  b:=d;
 end;
if b>c then
 begin
  d:=b;
  b:=c;
  c:=d;
 end;
begin
if a+b<c then writeln('Error!')
 else if a*a+b*b=c*c then writeln('Yes!')
  else writeln('No!');
end;
end;
end.
